src/ec/montgomery.rs

```diff
@@ -79,35 +82,39 @@ impl ObjectFactory for ECMontgomeryPubFactory {
         Ok(obj)
     }
 
-    fn get_attributes(&self) -> &Vec<ObjectAttr> {
-        &self.attributes
+    fn get_data(&self) -> &ObjectFactoryData {
+        &self.data
+    }
+
+    fn get_data_mut(&mut self) -> &mut ObjectFactoryData {
+        &mut self.data
     }
 }
 
 impl CommonKeyFactory for ECMontgomeryPubFactory {}
 
 impl PubKeyFactory for ECMontgomeryPubFactory {}
 
-#[derive(Debug)]
+#[derive(Debug, Default)]
 pub struct ECMontgomeryPrivFactory {
-    attributes: Vec<ObjectAttr>,
+    data: ObjectFactoryData,
 }
 
 impl ECMontgomeryPrivFactory {
     pub fn new() -> ECMontgomeryPrivFactory {
-        let mut data: ECMontgomeryPrivFactory = ECMontgomeryPrivFactory {
-            attributes: Vec::new(),
-        };
-        data.attributes.append(&mut data.init_common_object_attrs());
-        data.attributes
-            .append(&mut data.init_common_storage_attrs());
-        data.attributes.append(&mut data.init_common_key_attrs());
-        data.attributes
-            .append(&mut data.init_common_private_key_attrs());
-        data.attributes.push(attr_element!(
+        let mut factory: ECMontgomeryPrivFactory = Default::default();
+
+        factory.add_common_object_attrs();
+        factory.add_common_storage_attrs();
+        factory.add_common_key_attrs();
+        factory.add_common_private_key_attrs();
```

Comment:
Could the `add_common_private_key_attrs()` add all the above attributes at once without the need to call all 4 functions for each private key factory?

Reply:
I tried, but it doesn't work because each of these function is froma  different trait, and they do no derive from each other, lemme see if I can fix this with some clever trait magic